src/hardware/hantek-6xxx/api.c

```diff
@@ -707,6 +719,7 @@ static void LIBUSB_CALL receive_transfer(struct libusb_transfer *transfer)
 		devc->sample_buf = NULL;
 		sdi->driver->dev_acquisition_stop(sdi, NULL);
 	} else {
+		libusb_free_transfer(transfer);
```

Comment:
I tested it now but I could not find related memory leaks belonging to the driver. Copying the usb buffer logic from fx2lafw should be the "correct" reference.